MCS 320 Quiz 1 Friday 28 August 2026.

Consider $\displaystyle s = \sin\left(\sqrt{13}\right)$.

1. Compute a rational approximation for $s$, accurate to 3 decimal places.

2. Verify that the accuracy of your rational approximation is 3 decimal places.

   Write an explanation for your verification.

3. Compute the sequence of the first ten consecutive rational approximations for $s$,
   accurate respectively with 1, 2, up to 10 decimal places (10 included).

# Answer to Part 1

In [1]:
s = sin(sqrt(13))
show(s)

sin(sqrt(13))

In [2]:
a = QQ(s.n(digits=3))
show(a)

-183/409

# Answer to Part 2

In [3]:
absolute_error = abs(s-a).n(digits=4)
show(absolute_error)

0.00006485

In [4]:
relative_error = (absolute_error/abs(s)).n(digits=4)
show(relative_error)

0.0001449

In [5]:
'%.2e' % relative_error

'1.45e-04'

The relative error is $1.45 \times 10^{-4}$.  This verifies that the approximation has indeed the first 3 decimal places correct.

# Answer to Part 3

In [6]:
L = [QQ(s.n(digits=k)) for k in range(1, 11)]
show(L)

[-7/16,
 -17/38,
 -183/409,
 -179/400,
 -473/1057,
 -3247/7256,
 -13559/30300,
 -36761/82149,
 -142557/318569,
 -29027/64866]